sdk/core/System.ClientModel/README.md

```diff
@@ -23,71 +22,290 @@ dotnet add package System.ClientModel
+
+    string IPersistableModel<SampleResource>.GetFormatFromOptions(ModelReaderWriterOptions options)
+        => options.Format;
+
+    void IJsonModel<SampleResource>.Write(Utf8JsonWriter writer, ModelReaderWriterOptions options)
+        => ToJson(writer);
+
+    BinaryData IPersistableModel<SampleResource>.Write(ModelReaderWriterOptions options)
+        => ModelReaderWriter.Write(this, options);
+
+    // Write the model JSON that will populate the HTTP request content.
+    private void ToJson(Utf8JsonWriter writer)
+    {
+        writer.WriteStartObject();
+        writer.WritePropertyName("id");
+        writer.WriteStringValue(Id);
+        writer.WriteEndObject();
+    }
+
+    // Read the JSON response content and create a model instance from it.
+    private static SampleResource FromJson(Utf8JsonReader reader)
+    {
+        reader.Read(); // start object
+        reader.Read(); // property name
+        reader.Read(); // id value
+
+        return new SampleResource(reader.GetString()!);
+    }
+}
+```
+
+For more information on reading and writing persistable models, see [Model reader writer samples](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/System.ClientModel/samples/ModelReaderWriter.md).
+
+### Accessing HTTP response details
+
+Service clients have methods that are used to call cloud services to invoke service operations.  These methods on a client are called **service methods**. Service clients expose two types of service methods: [convenience methods](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/System.ClientModel/samples/ServiceMethods.md#convenience-methods) and [protocol methods](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/System.ClientModel/samples/ServiceMethods.md#protocol-methods).
+
+**Convenience methods** provide a convenient way to invoke a service operation.  They are methods that take a strongly-typed model as input and return a `ClientResult<T>` that holds a strongly-typed representation of the service response.  Details from the HTTP response may also be obtained from the return value.
 
-A very basic client implementation might use the following approach:
+**Protocol method** are low-level methods that take parameters that correspond to the service HTTP API and return a `ClientResult` holding only the raw HTTP response details.  These methods also take an optional `RequestOptions` parameter that allows the client pipeline and the request to be configured for the duration of the call.
 
-```csharp
-ApiKeyCredential credential = new ApiKeyCredential(key);
-ApiKeyAuthenticationPolicy authenticationPolicy = ApiKeyAuthenticationPolicy.CreateBearerAuthorizationPolicy(credential);
-ClientPipeline pipeline = ClientPipeline.Create(pipelineOptions, authenticationPolicy);
+The following sample illustrates how to call a convenience method and access both the strongly-typed output model and the details of the HTTP response.
```

Comment:
This sample looks a bit too busy/complex for something called "convenience method". It's hard to discern what is continent about it. Should we remove getting the key and creds, accessing the raw response, etc. Also, to understand what's a convinience method, people need to understand what a protocol method is, but I don't think we should put protocol method first, as it might create an impression that in mainline scenarios users need to use protocol methods.

Reply:
I can isolate this to a single call to make it look more convenient :).  I have a discussion of convenience/protocol methods higher up, but I can include a short recap of that here for people who jump straight to this section

Reply:
Addressed